### PR TITLE
Modify EngineProgram.get_circuit() to work with BatchPrograms

### PR DESCRIPTION
**Pull request by dstrain115**

### CODE REVIEW

File: cirq/google/engine/engine_program.py [hunk 1]
```diff
@@ -387,28 +387,45 @@ def remove_labels(self, keys: List[str]) -> 'EngineProgram':
...
+        program = None
         if (code_type == 'cirq.google.api.v2.Program' or
                 code_type == 'cirq.api.google.v2.Program'):
             program = v2.program_pb2.Program.FromString(code.value)
+        if code_type == 'cirq.google.api.v2.BatchProgram':
+            if program_num is None:
+                raise ValueError('A program number must be specified when '
+                                 'deserializing a Batch Program')
+            batch = v2.batch_pb2.BatchProgram.FromString(code.value)
+            if program_num >= len(batch.programs):
```
Comment (mrwojtek) [inline cirq/google/engine/engine_program.py:423]: The
if math.abs(program_num) >= len(batch.programs)
would be more appropriate

File: cirq/google/engine/engine_program.py [hunk 2]
```diff
@@ -387,28 +387,45 @@ def remove_labels(self, keys: List[str]) -> 'EngineProgram':
...
         import cirq.google.engine.engine as engine_base
         code_type = code.type_url[len(engine_base.TYPE_PREFIX):]
         if (code_type == 'cirq.google.api.v1.Program' or
                 code_type == 'cirq.api.google.v1.Program'):
             raise ValueError('deserializing a v1 Program is not supported')
+        program = None
         if (code_type == 'cirq.google.api.v2.Program' or
                 code_type == 'cirq.api.google.v2.Program'):
             program = v2.program_pb2.Program.FromString(code.value)
+        if code_type == 'cirq.google.api.v2.BatchProgram':
```
Comment (mrwojtek) [inline cirq/google/engine/engine_program.py:418]: optional: elif instead of if

**Review by mrwojtek — APPROVED** [review]

File: cirq/google/engine/engine_program.py (hunk 1 above)
Comment (dstrain115) [reply cirq/google/engine/engine_program.py]: Done.

File: cirq/google/engine/engine_program.py (hunk 2 above)
Comment (dstrain115) [reply cirq/google/engine/engine_program.py]: Done.

**Pull request merged**